manifests/key_pair.pp

```diff
@@ -0,0 +1,32 @@
+define certs::key_pair (
+  $source_dir,
+  $key_file,
+  $cert_file,
+  $key_owner   = undef,
+  $key_group   = undef,
+  $key_mode    = undef,
+  $cert_owner  = undef,
+  $cert_group  = undef,
+  $cert_mode   = undef,
+  $require     = undef,
```

Comment:
I don't think you need to do this. That should be built into Puppet itself already.